apps/prairielearn/src/lib/db-types.ts

```diff
@@ -9,10 +9,27 @@ export const IdSchema = z
   .string({ coerce: true })
   .refine((val) => /^\d+$/.test(val), { message: 'ID is not an integer' });
 
+// Accepts either a string or a Date object. If a string is passed, it is
+// validated and parsed as an ISO date string.
+//
+// Useful for parsing dates from JSON, which are always strings.
+export const DateFromISOString = z
+  .union([z.string(), z.date()])
+  .refine(
+    (s) => {
+      const date = new Date(s);
+      return !Number.isNaN(date.getTime());
+    },
+    {
+      message: 'must be a valid ISO date string',
+    },
+  )
+  .transform((s) => new Date(s));
```

Comment:
I pulled this over from PT and modified it so that it would also accept `Date` objects, which we would get if we aren't using `to_jsonb` in our queries. I can lift this into a separate PR if desired.

Reply:
Nice! No need for a separate PR.